### PR TITLE
fill result from history

### PR DESCRIPTION
**Pull request by plakrisenko**
read_result_from_file updated to collect results from multiple history files

### CODE REVIEW

**Review by yannikschaelte — COMMENTED** [review]
@FFroehlich before reviewing: I think `read_result_from_file` was from you?
I guess it would be preferable to keep both singular and plural versions.

File: pypesto/optimize/optimizer.py [hunk 1]
```diff
@@ -197,33 +200,71 @@ def fill_result_from_objective_history(
...
+        )
+
+        result = OptimizerResult(
+            id=identifier,
+            message='loaded from file',
+            exitflag=EXITFLAG_LOADED_FROM_FILE,
+            time=max(history.get_time_trace())
+        )
+        result.id = identifier
+        result = fill_result_from_objective_history(result, opt_hist)
```
Comment (yannikschaelte) [inline pypesto/optimize/optimizer.py:262]: The repeated execution here will not work properly yet, as it won't take the minimum, but at the moment rather only performs one replacement.

**Comment by FFroehlich** [comment]
> @FFroehlich before reviewing: I think `read_result_from_file` was from you?
> I guess it would be preferable to keep both singular and plural versions.

Correct. Ideally `read_result_from_files` does some comprehension using `read_result_from_file` and then performs merging. Would also yield smaller functions.

**Comment by yannikschaelte** [comment]
> > @FFroehlich before reviewing: I think `read_result_from_file` was from you?
> > I guess it would be preferable to keep both singular and plural versions.
> 
> Correct. Ideally `read_result_from_files` does some comprehension using `read_result_from_file` and then performs merging. Would also yield smaller functions.

Exactly. I think merging the results list to a OptimizeResult, as currently done, is good. So @plakrisenko , could you make the plural function loop over and merge the singular function? Then that should be fine.

**Pull request closed without merging**